From db8d0f4c14ba9f8682c2543e920d0dc04d80b1c2 Mon Sep 17 00:00:00 2001
From: tropicrainforest <tropicrainforest27017@gmail.com>
Date: Sun, 7 Jun 2026 16:54:33 +0200
Subject: [PATCH 18/28] ssl_types: make SSLX509Name::Entry operator== a member

The equality comparison for SSLX509Name::Entry was a free (non-member) function,
and its presence pulled the surrounding Entry container into instantiation while
the type was still incomplete (clang-19 / libc++ 22). Converting it into a
member function defers that until Entry is complete.
---
 src/mongo/util/net/ssl_types.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/mongo/util/net/ssl_types.h b/src/mongo/util/net/ssl_types.h
index bb65d271..284e07dd 100644
--- a/src/mongo/util/net/ssl_types.h
+++ b/src/mongo/util/net/ssl_types.h
@@ -61,6 +61,9 @@ public:
         auto equalityLens() const {
             return std::tie(oid, type, value);
         }
+        bool operator==(const Entry& rhs) const {
+            return this->equalityLens() == rhs.equalityLens();
+        }
     };
 
     SSLX509Name() = default;
@@ -110,9 +113,6 @@ private:
 };
 
 std::ostream& operator<<(std::ostream&, const SSLX509Name&);
-inline bool operator==(const SSLX509Name::Entry& lhs, const SSLX509Name::Entry& rhs) {
-    return lhs.equalityLens() == rhs.equalityLens();
-}
 inline bool operator<(const SSLX509Name::Entry& lhs, const SSLX509Name::Entry& rhs) {
     return lhs.equalityLens() < rhs.equalityLens();
 }
